### PR TITLE
Write document about environment variables

### LINKED ISSUE #1227: Write a document about environment variables
It is too hard to find environment variables Chainer uses. We need a page about that.

- `CHAINER_CUDNN`
- `CHAINER_SEED`
- `CHAINER_TYPE_CHECK`
- `CUPY_CACHE_DIR`

for install
- `CUDA_PATH`

for test
- `CUPY_TEST_FULL_COMBINATION`

### PR DESCRIPTION
**Pull request by unnonouno**
fix #1227

We need to consider about CHAINER_SEED.

### CODE REVIEW

File: docs/source/reference/environment.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+=====================
+
+Here are environment variables Chainer uses.
+
+
++------------------------+----------------------------------------------------------+
+| ``CHAINER_CUDNN``      | Set ``0`` to disable cuDNN in Chainer.                   |
+|                        | Otherwise cuDNN is enabled automatically.                |
++------------------------+----------------------------------------------------------+
+| ``CHAINER_SEED``       | Default seed value of random number generators for CUDA. |
```
Comment (zori) [inline docs/source/reference/environment.rst]: If the variable is not set, what value does Chainer use?

File: docs/source/reference/environment.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+|                        | Set an integer value in decimal format.                  |
++------------------------+----------------------------------------------------------+
+| ``CHAINER_TYPE_CHECK`` | Set ``0`` to disable type checking.                      |
+|                        | Otherwise type checking is enabled automatically.        |
+|                        | See :class:`~chainer.Function` for detail.               |
```
Comment (zori) [inline docs/source/reference/environment.rst]: for details.

File: docs/source/cupy-reference/environment.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+| ``CUPY_CACHE_DIR`` | Path to the directory to store kernel cache.       |
+|                    | ``$(HOME)/.cupy.kernel_cache`` is used by default. |
+|                    | See :ref:`cupy-overview` for detail.               |
++--------------------+----------------------------------------------------+
+
+
+For install
+-----------
+
+These environment variables are only referred on intall.
```
Comment (zori) [inline docs/source/cupy-reference/environment.rst]: ... only used during installation.

File: docs/source/cupy-reference/environment.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
++---------------+----------------------------------------+
+| ``CUDA_PATH`` | Path to the directory containing CUDA. |
+|               | See :ref:`install_cuda` for detail     |
```
Comment (zori) [inline docs/source/cupy-reference/environment.rst]: for details.

File: docs/source/cupy-reference/environment.rst [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
++--------------------+----------------------------------------------------+
+
+
+For install
+-----------
+
+These environment variables are only referred on intall.
+
++---------------+----------------------------------------+
+| ``CUDA_PATH`` | Path to the directory containing CUDA. |
```
Comment (zori) [inline docs/source/cupy-reference/environment.rst]: The default would be the directory where `nvcc` is, right? Could you please note that.

File: docs/source/cupy-reference/environment.rst [hunk 6]
```diff
@@ -0,0 +1,22 @@
+Environment variables
+=====================
+
+Here are environment variables Chainer uses.
```
Comment (zori) [inline docs/source/cupy-reference/environment.rst]: are _the_ environment variables

File: docs/source/reference/environment.rst [hunk 7]
```diff
@@ -0,0 +1,17 @@
+Environment variables
+=====================
+
+Here are environment variables Chainer uses.
```
Comment (zori) [inline docs/source/reference/environment.rst]: ditto (are _the_ environment variables)

**Comment by zori** [comment]
Also maybe explain about `CUPY_TEST_FULL_COMBINATION` which is mentioned in https://github.com/pfnet/chainer/issues/1227

**Comment by zori** [comment]
LGTM, except that I'm not clear on `CUPY_TEST_FULL_COMBINATION`: should we ignore it for now?

**Comment by unnonouno** [comment]
I didn't write it because this environment variable is not for users but developers. How do you think?

**Comment by zori** [comment]
Makes sense. LGTM.

**Pull request merged**